### ISSUE TITLE
Windows - `customFlags` should use `filepath.Clean()` instead of `path.Clean()`

### ISSUE THREAD

**Issue by tintinweb** [ethereum/go-ethereum#22367]
`go-ethereum/cmd/utils/customflags.go` is handling OS paths and therefore it should use `filepath.Clean()` (works for windows path separator) instead of `path.Clean()` (nix path separator)

https://github.com/ethereum/go-ethereum/blob/fdb742419e6ff8373e57bdf99836c22f689b0e55/cmd/utils/customflags.go#L199-L199

`path.Clean()`
https://github.com/golang/go/blob/master/src/path/path.go#L51-L62

`filepath.Clean()`
https://github.com/golang/go/blob/master/src/path/filepath/path.go#L66-L79

#### Expected behaviour

should cleanup for OS path separator

#### Actual behaviour

`path.Clean()` assumes the path separator is `/` while it might be `\\`

**Comment by karalabe** [comment]
Good catch. Though we'd need to update that sole test too to pass on Windows if we update the test. I assume it will break because it explicitly hard codes Linux paths and expansions.
